File: Components/WinForms/CS/TreeListTutorials/CustomTreeListNode/CustomTreeListNode.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace DevExpress.XtraTreeList.Demos.Tutorials {
    /// <summary>
    /// Summary description for CustomTreeListNode.
    /// </summary>
    public partial class CustomTreeListNode : TutorialControl {
        public CustomTreeListNode() {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();
            treeList1.AppendNode(null, null);
            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }
        public override bool SetNewWhatsThisPadding { get { return true; } }
        private void treeList1_FocusedNodeChanged(object sender, DevExpress.XtraTreeList.FocusedNodeChangedEventArgs e) {
            button2.Enabled = button3.Enabled = e.Node != null;
        }

        private void button1_Click(object sender, System.EventArgs e) {
            treeList1.FocusedNode = treeList1.AppendNode(null, null);
        }

        private void button2_Click(object sender, System.EventArgs e) {
            if(treeList1.FocusedNode != null)
                treeList1.FocusedNode = treeList1.AppendNode(null, treeList1.FocusedNode);
        }

        private void button3_Click(object sender, System.EventArgs e) {
            if(treeList1.FocusedNode != null)
                treeList1.DeleteNode(treeList1.FocusedNode);
        }
    }
}

File: Components/WinForms/VB/TreeListTutorials/CustomTreeListNode/CustomTreeListNode.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data

Namespace DevExpress.XtraTreeList.Demos.Tutorials
	''' <summary>
	''' Summary description for CustomTreeListNode.
	''' </summary>
	Public Partial Class CustomTreeListNode
		Inherits TutorialControl
		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
            InitializeComponent()
            Dim aNode As DevExpress.XtraTreeList.Nodes.TreeListNode = Nothing
            treeList1.AppendNode(Nothing, aNode)
			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub
        Public Overrides ReadOnly Property SetNewWhatsThisPadding() As Boolean
            Get
                Return True
            End Get
        End Property
		Private Sub treeList1_FocusedNodeChanged(ByVal sender As Object, ByVal e As DevExpress.XtraTreeList.FocusedNodeChangedEventArgs) Handles treeList1.FocusedNodeChanged
            button3.Enabled = Not e.Node Is Nothing
			button2.Enabled = button3.Enabled
		End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
            Dim aNode As DevExpress.XtraTreeList.Nodes.TreeListNode = Nothing
            treeList1.FocusedNode = treeList1.AppendNode(Nothing, aNode)
        End Sub

		Private Sub button2_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button2.Click
			If Not treeList1.FocusedNode Is Nothing Then
				treeList1.FocusedNode = treeList1.AppendNode(Nothing, treeList1.FocusedNode)
			End If
		End Sub

		Private Sub button3_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button3.Click
			If Not treeList1.FocusedNode Is Nothing Then
				treeList1.DeleteNode(treeList1.FocusedNode)
			End If
		End Sub
	End Class
End Namespace
